### DISCUSSION TITLE
I want to enable or disable the wifi, is there a way to do this?

### DISCUSSION THREAD

**Discussion by danielpaulus** [danielpaulus/go-ios#90] [Q&A]
The only way I can think of is by building a small xcuitest for that. You could run it easily with go-ios.
Not the greatest solution, i know :-D

**Comment by wellkilo** [comment]
go-ios still does not expose a command that toggles the device's Wi-Fi radio on or off.

The current ios wifi command is different: it installs or removes a managed Wi-Fi network profile:

    ios wifi --ssid='MyNetwork' --password='secret' --enc-type=WPA
    ios wifi --ssid='MyNetwork' --remove

That can make the device join/forget a configured network, but it does not toggle Wi-Fi hardware. On an unsupervised device, installing/removing the profile may also require confirmation under Settings → General → VPN & Device Management; supervised/setup devices can apply it silently. This behavior is implemented in [wifi.go](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/ios/mcinstall/wifi.go#L14-L71) and documented by the CLI help in [main.go](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/main.go#L442-L444).

For a true on/off switch, XCUITest/DeviceKit UI automation remains the practical route unless Apple exposes an appropriate managed-device command for your supervision/MDM scenario. Be aware that UI automation against Settings is OS-version/language dependent, so locate controls by accessibility identifiers where possible rather than screen coordinates.
